### PR TITLE
edit readme

### PR DESCRIPTION
**Pull request by zaraG**
add notice of adding symlink for mac users

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -19,6 +19,10 @@ Use trial to run the test suite::
 
   trial leap.keymanager
 
+Note: If you are on mac, make sure you have /usr/bin/gpg either as an executable or a symlink to it.
+
+  ln -s [physical path] /usr/bin/gpg
+
```
Comment (meskio) [inline README.rst]: In theory it should find the binary of the gpg executable where ever it is, if it's in the $PATH:
https://github.com/leapcode/keymanager/blob/develop/src/leap/keymanager/tests/__init__.py#L92

I'm not sure if we need this note at all, but if you want it there I would like it to be more generic, this problem is not related to mac, but to any platform, and not only related to the tests.

To use KeyManager you need to have gpg installed. I don't think is part of this README to explain how to configure your environment, set $PATH or install gpg. But I will be happy to mention the need of having gpg installed.

**Comment by zaraG** [comment]
Some how it didn't worked on some mac's, that is why we added it in the README not just to install gpg, but also to set the symlink, to make it easier to contribute. But I don't know if it is really "needed" and what is the best place to put it (wiki?).

**Comment by bwagnerr** [comment]
hey meskio, I have a fix accepted but not released yet on isis' gnupg to fix the problem with finding the gpg executable on mac. We documented this specific case because it always happens on mac, on most linuxes py gnupg finds the gpg correctly. I would like to have this note on the README while we don't have a new release of python gnupg

**Comment by kalikaneko** [comment]
I wonder if the symlink actually works:
https://python-gnupg.readthedocs.io/en/latest/gnupg.html#gnupg._util._find_binary

**Comment by bwagnerr** [comment]
@kalikaneko For me it shouldn't, but it does.

**Comment by zaraG** [comment]
For some other ppl in the office it is the same behaviour as @bwagnerr, but when I try to directly test a method which uses the `TempGPGWrapper()` it still says 
> exceptions.RuntimeError: GnuPG is not installed!

I wonder how that happens. Any suggestions?

**Comment by felixhammerl** [comment]
BTW: for those on mac os, this may be necessary:

* Boot OS X into Recovery Mode: Reboot and Hold down Command + R keys simultaneously after you hear the startup chime
* In Revocery mode, open the Terminal from the Utilities menu at the top of the screen
* Disable System Integrity Protection and reboot: `$csrutil disable; reboot`

**Comment by zaraG** [comment]
We faced some other problems if you want to set up the environment from scratch (in a virtuelnv to use ubuntu), I am currently trying to improve the [pip_install_requirements.sh](https://github.com/pixelated/keymanager/blob/develop/pkg/pip_install_requirements.sh), that is what we did so far..just to document it somewhere.. :)

sudo apt-get install git
sudo apt-get install python-pip
sudo pip install -U pip

Create Virtualenvironment
sudo pip install virtualenv
virtualenv keymanager -p /usr/bin/python2.7
source keymanager/bin/activate

cd /vagrant/
python setup.py develop

sudo apt-get install build-essential libssl-dev libffi-dev python-dev

pip install -r  pkg/requirements-testing.pip
pip install -r  pkg/requirements-latest.pip
python setup.py test

**Comment by bwagnerr** [comment]
Hey @zaraG, I was more in favor of removing any sh for bootstrap, setup.py or pip should be able to handle whatever we need, they are made to do the bootstraping anyway.

With this in mind, I would add a note on the README about the needed libraries, something along the lines:

Requirements

To build the keymanager you need gcc, libssl, libffi, python-dev

On a debian based machine, you can run this
```
apt-get install build-essential libssl-dev libffi-dev python-dev
```


The rest should not be given (using a virtualenv, the path itself)
Also in your steps, the python setup.py develop should not be needed, the pip installs cover for it.

Could you tweak the README with these changes?

**Comment by zaraG** [comment]
@bwagnerr I agree with you with the virtaulenv stuff, I just paste everything what I did, but for some reason I needed to run the requirements (latest and testing) as well, even if they are already  called in the setup.py..I wonder about that..

I also edited the the requirements-latest.pip, because it was complaining about missing `dirspec` .. and it is cloning from [pixelated-project instead of leapcode](https://github.com/leapcode/keymanager/blob/develop/pkg/requirements-latest.pip). 

Feel a bit lost how to make it properly. So that people won't face the same problem we faced yesterday.

**Comment by bwagnerr** [comment]
Sorry, I meant the other way around. You could have used only pip (and never ran setup.py develop), the two requirements are needed to install the keymanager in develop mode.

The requirements-latest always pointed for the latest pix code, considering the latest development, I think it should point to the latest leap develop.

Maybe we should wait for the development of #113, because that is meant to make the setup process simpler, that should help :)

**Comment by zaraG** [comment]
Ok, that looks pretty good, but it still points to the pixelated code instead of leap, right?
Probably that is no problem, bc they are supposed to be in sinc, right ;)

**Pull request closed without merging**